src/msg/Message.h

```diff
@@ -221,6 +225,13 @@
 // abstract Message class
 
 class Message : public RefCountedObject {
+public:
+#ifdef WITH_SEASTAR
+  using ConnectionRef = ceph::net::SocketConnectionRef;
```

Comment:
`using ConnectionRef = ceph::net::ConnectionRef;` would be better.

This complies with the convention of classic messenger:
`SocketConnectionRef` is supposed to be internal API of messenger, because it exposes unnecessary interfaces to users.

Reply:
The issue is that e.g. the freshly added `get_peer_socket_addr()` requires `target_addr` which is a part of `ceph::net::SocketConnection`, not `ceph::not::Connection`. We could make it virtual but there might be a simpler solution: how do you feel about migrating `target_addr` to crimson's `Connection`?